mesonbuild/backend/ninjabackend.py

```diff
@@ -1116,12 +1119,13 @@ def generate_dependency_scan_target(self, target: build.BuildTarget, compiled_so
         for g in generated_source_files:
             elem.orderdeps.add(g.relative_name())
         elem.orderdeps.update(object_deps)
-        scaninfo = TargetDependencyScannerInfo(self.get_target_private_dir(target), source2object)
+        scaninfo = TargetDependencyScannerInfo(
+            self.get_target_private_dir(target), source2object, scan_sources)
```

Comment:
This is not a missing function annotation, it's just accidentally committed in the wrong commit.